src/main/kotlin/me/serce/solidity/lang/resolve/engine.kt

```diff
@@ -322,13 +322,44 @@ object SolResolver {
         ?.supers
         ?.flatMap { resolveTypeNameUsingImports(it) }
         ?: emptyList()
-      else -> lexicalDeclarations(element)
-        .filter { it.name == element.name }
-        .toList().let {
-          if (it.size <= 1 || it.any { it !is SolContractDefinition }) it
-          // resolve by imports to distinguish elements with the same name
-          else resolveTypeNameUsingImports(element).toList()
+      else -> {
+        //check if the element is an alias
+        var nameToFilter = element.name
+        run {
+          element.containingFile.childrenOfType<SolImportDirective>().forEach {
+            if (it.importAlias?.name == element.name) {
+              //match import a as A from "path"
+              if (it.userDefinedTypeName != null) {
+                nameToFilter = it.userDefinedTypeName!!.name
+                return@run
+              } else {
+              //match import "path" as A
+                val contractFromPath =
+                  it.importPath?.reference?.resolve()?.childrenOfType<SolContractDefinition>()
+                if (!contractFromPath.isNullOrEmpty()) {
+                  nameToFilter = contractFromPath[0].name
```

Comment:
why do we take the first contract only? there may be multiple contracts in a file